g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/v2/about/AboutVersionAPI.java

```diff
@@ -44,6 +45,11 @@
     public APIResponseDTO getVersionInfo(
             @ApiParam(name = APIParamName.PRODUCT_NAME, required = false, value = APIParamValue.PRODUCT_NAME) @RequestParam(required = false) String productName,
             @ApiParam(name = APIParamName.VERSION, required = false, value = APIParamValue.VERSION) @RequestParam(required = false) String version) throws AboutAPIException {
-        return super.handleResponse(APIResponseStatus.OK, versionService.getBuildVersion(productName, version));
+        if(!StringUtils.isEmpty(productName) && !StringUtils.isEmpty(version)) {
+            String availableVersion = super.getAvailableVersion(productName, version);
+            return super.handleVersionFallbackResponse(version, availableVersion, versionService.getBuildVersion(productName, availableVersion));
+        }else{
+            return super.handleResponse(APIResponseStatus.OK, versionService.getBuildVersion(productName, version));
```

Comment:
In this block, there must be this case: either productname or version is empty. How to get build version in this case?

Reply:
Only return service version.

Reply:
After change it will return error message.